components/eam/src/physics/cam/tropopause.F90

```diff
@@ -476,49 +385,48 @@ subroutine tropopause_climate(pstate, tropLev, tropP, tropT, tropZ)
         last = 12
         dels = (calday - days(12)) / (365._r8 + days(1) - days(12))
       else
-        do m = 11,1,-1
-           if( calday >= days(m) ) then
+        do mn = 11,1,-1
+           if( calday >= days(mn) ) then
               exit
-           end if
-        end do
-        last = m
-        next = m + 1
-        dels = (calday - days(m)) / (days(m+1) - days(m))
+           endif
+        enddo
+        last = mn
+        next = mn + 1
+        dels = (calday - days(mn)) / (days(mn+1) - days(mn))
       end if
       
-      dels = max( min( 1._r8,dels ),0._r8 )
-        
+      dels = min_max_bound(0._r8,1._r8,dels)  
 
       ! Iterate over all of the columns.
-      do i = 1, ncol
+      do icol = 1, ncol
        
         ! Skip column in which the tropopause has already been found.
-        if (tropLev(i) == NOTFOUND) then
+        if (tropLev(icol) == NOTFOUND) then
         
         !--------------------------------------------------------
         ! ... get tropopause level from climatology
         !--------------------------------------------------------
           ! Interpolate the tropopause pressure.
-          tP = tropp_p_loc(i,lchnk,last) &
-            + dels * (tropp_p_loc(i,lchnk,next) - tropp_p_loc(i,lchnk,last))
+          tP = tropp_p_loc(icol,lchnk,last) &
+            + dels * (tropp_p_loc(icol,lchnk,next) - tropp_p_loc(icol,lchnk,last))
```

Comment:
This part of the code uses the data (read from a file) based on the location of `icol` (or column) on the globe. This needs to be modified based on how we read files on the EAMxx side. Let's leave a note here (e.g., "`tropp_p_loc` has location-based data as a function of `icol` and `lchnk`, C++ ported code needs to read in the data and use the data based on the column it is operating on") and we need to address it in the ported C++ code.

Reply:
OK, done, thanks.